packages/cli/src/UserManagement/PermissionChecker.ts

```diff
@@ -135,14 +129,7 @@ export class PermissionChecker {
 		}
 
 		if (policy === 'workflowsFromSameOwner') {
-			const user = await Container.get(UserService).findOne({ where: { id: userId } });
-			if (!user) {
-				throw new WorkflowOperationError(
-					'Fatal error: user not found. Please contact the system administrator.',
-				);
-			}
-			const sharing = await WorkflowsService.getSharing(user, subworkflow.id, ['role', 'user']);
-			if (!sharing || sharing.role.name !== 'owner') {
+			if (!subworkflowOwner || subworkflowOwner.id !== userId) {
```

Comment:
Should be equivalent?

```suggestion
			if (subworkflowOwner?.id !== userId) {
```